File: src/dataset/cpp/beer-song_exercism-tqa236/gemini/GeminiFlash_beer_song_v2.cpp
#include "beer_song.h"

#include <sstream>

std::string beer_song::verse(int verse_num) {
  if (verse_num == 0) {
    return "No more bottles of beer on the wall, no more bottles of beer.\nGo "
           "to the store and buy some more, 99 bottles of beer on the wall.\n";
  }

  std::stringstream ss;
  if (verse_num == 1) {
    ss << verse_num << " bottle of beer on the wall, " << verse_num
       << " bottle of beer.\nTake it down and pass it around, no more bottles "
          "of beer on the wall.\n";
  } else {
    ss << verse_num << " bottles of beer on the wall, " << verse_num
       << " bottles of beer.\nTake one down and pass it around, "
       << verse_num - 1 << (verse_num == 2 ? " bottle" : " bottles")
       << " of beer on the wall.\n";
  }
  return ss.str();
}

std::string beer_song::sing(int start, int end) {
  std::stringstream lyrics;
  for (int i = start; i > end - 1; --i) {
    lyrics << beer_song::verse(i);
    if (i != end) {
      lyrics << "\n";
    }
  }
  return lyrics.str();
}

std::string beer_song::sing(int start) { return beer_song::sing(start, 0); }

File: src/dataset/javascript/beer-song_exercism-elenamountz/gemini/GeminiFlash_beer-song_v2.js
export default class Beer {
  constructor() {}

  static verse(n) {
    switch (n) {
      case 0:
        return "No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n";
      case 1:
        return "1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n";
      case 2:
        return "2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n";
      default:
        return `${n} bottles of beer on the wall, ${n} bottles of beer.\nTake one down and pass it around, ${n - 1} bottles of beer on the wall.\n`;
    }
  }

  static sing(from = 99, to = 0) {
    let result = "";
    for (let i = from; i >= to; i--) {
      result += Beer.verse(i) + "\n";
    }
    return result.slice(0, -1);
  }
}